core/roboconf-agent/src/main/java/net/roboconf/agent/internal/misc/AgentUtils.java

```diff
@@ -112,6 +118,39 @@ public static void copyInstanceResources( Instance instance, Map<String,byte[]>
 
 
 	/**
+	 * Executes a script resource on a given instance.
+	 * @param instance an instance
+	 * @throws InterruptedException
+	 * @throws IOException
+	 */
+	public static List<ExecutionResult> executeScriptResources( Instance instance, String applicationName, String scopedInstanceName ) throws IOException, InterruptedException {
+
+		File dir = InstanceHelpers.findInstanceDirectoryOnAgent( instance );
+		Utils.createDirectory(dir);
+		List<File> scriptFiles = Utils.listAllFiles( dir, Constants.FILE_EXT_SCRIPT);
+		List<ExecutionResult> results = new ArrayList<ExecutionResult> ();
+		Logger logger = Logger.getLogger( AgentUtils.class.getName());
+
+		for( File script : scriptFiles) {
+			StringBuilder normalOutput = new StringBuilder();
+			StringBuilder errorOutput = new StringBuilder();
+			ExecutionResult result = new ExecutionResult(normalOutput.toString(),errorOutput.toString(), -1);
+			if( script.exists()) {
+				if(! script.canExecute())
+					script.setExecutable(true);
+
+				String[] command = { script.getAbsolutePath() };
+				result = ProgramUtils.executeCommandWithResult( logger, command, script.getParentFile(), null, applicationName, scopedInstanceName);
+			} else {
+				logger.severe("There is no script to execute");
```

Comment:
No, a script is not mandatory, no need to output a severe log when there is no script. No need to log anything in fact.